src/internal/pfsdb/branches.go

```diff
@@ -213,7 +213,7 @@ func GetBranchInfoWithID(ctx context.Context, tx *pachsql.Tx, b *pfs.Branch) (*B
 	if err := tx.GetContext(ctx, row, getBranchByNameQuery, project, repo, repoType, branch); err != nil {
 		if err == sql.ErrNoRows {
 			if _, err := GetRepoByName(ctx, tx, project, repo, repoType); err != nil {
-				return nil, errors.Join(err, &BranchNotFoundError{BranchKey: b.Key()})
+				return nil, errors.Wrapf(err, "get repo for branch info %v", b.Key())
```

Comment:
The reason we join is that we don't want to lose information about the error. An error in GetBranch is still an `branch not found` if the repo is not found as well.

I suggest we first check for if the error is a repo not found.
```suggestion
if errors.As(err, new(*RepoNotFoundError)) {
    return nil, errors.Join(err, &BranchNotFoundError{BranchKey: b.Key()})
}
return nil, errors.Wrapf(err, "get repo for branch info %v", b.Key())
```

Reply:
should be updated to match the suggestion in each place.